### PR TITLE
cradox is not supported anymore.

### PR DESCRIPTION
**Pull request by mrunge**
Update the docs etc. to use rados directly.

### CODE REVIEW

**Review by jd — APPROVED** [review]

**Review by chungg — APPROVED** [review]
will force merge this later.

**Comment by mrunge** [comment]
Thank you.

**Comment by tobias-urdin** [comment]
lets merge this now t hat CI is passing

**Comment by chungg** [comment]
@Mergifyio rebase

**Comment by chungg** [comment]
i tried this using latest image and it could be an issue with cradox and whatever ceph version comes with Bionic. i got:
```
{0} gnocchi.tests.test_indexer.TestIndexerDriver.test_delete_resource [60.964418s] ... FAILED

Captured traceback:
~~~~~~~~~~~~~~~~~~~
    b'Traceback (most recent call last):'
    b'  File "/home/tester/gnocchi/gnocchi/incoming/__init__.py", line 117, in NUM_SACKS'
    b'  File "/home/tester/gnocchi/gnocchi/incoming/ceph.py", line 70, in _get_storage_sacks'
    b'  File "cradox.pyx", line 499, in cradox.requires.wrapper.validate_func'
    b'  File "cradox.pyx", line 2695, in cradox.Ioctx.read'
    b"cradox.TimedOut: [errno 110] Ioctx.read(b'`\\xb4e\\xd2\\xee\\x7f'): failed to read b'gnocchi-config'"
    b''
    b'During handling of the above exception, another exception occurred:'
    b''
    b'Traceback (most recent call last):'
    b'  File "/home/tester/gnocchi/gnocchi/incoming/__init__.py", line 127, in upgrade'
    b'  File "/home/tester/gnocchi/gnocchi/incoming/__init__.py", line 119, in NUM_SACKS'
    b"gnocchi.incoming.SackDetectionError: [errno 110] Ioctx.read(b'`\\xb4e\\xd2\\xee\\x7f'): failed to read b'gnocchi-config'"
    b''
    b'During handling of the above exception, another exception occurred:'
    b''
    b'Traceback (most recent call last):'
    b'  File "/home/tester/gnocchi/gnocchi/tests/base.py", line 60, in skip_if_not_implemented'
    b'  File "/home/tester/gnocchi/gnocchi/tests/base.py", line 373, in setUp'
    b'  File "/home/tester/gnocchi/gnocchi/incoming/__init__.py", line 129, in upgrade'
    b'  File "/home/tester/gnocchi/gnocchi/incoming/ceph.py", line 74, in set_storage_settings'
    b'  File "cradox.pyx", line 499, in cradox.requires.wrapper.validate_func'
    b'  File "cradox.pyx", line 2626, in cradox.Ioctx.write_full'
    b"cradox.TimedOut: [errno 110] Ioctx.write_full(b'\\x01'): failed to write b'gnocchi-config'"
    b''
```

don't know how to make tox use libraries not from pypi but i think we need to use python3-rados

**Comment by mrunge** [comment]
cradox is not supported, see https://github.com/sileht/pycradox

Unfortunately, I haven't found a rados package in pypi, there are distribution packages though, like the before mentioned python3-rados

**Comment by mrunge** [comment]
Is there anything I can do to help getting this merged?

**Comment by jd** [comment]
Looks like travis did not run. :(

**Comment by jd** [comment]
Anyone knows why Travis is lazy on this one?

File: .travis.yml
Comment (tobias-urdin) [inline .travis.yml:51]: This doesn't have any effect, please remove this. The package is maintained inside the Docker container used for testing.

**Review by tobias-urdin — CHANGES_REQUESTED** [review]
Let's hold off with this until the CI fix for Bionic is merged as well then rebase this.

**Pull request merged**
